dev/com.ibm.ws.microprofile.reactive.messaging_fat/fat/src/com/ibm/ws/microprofile/reactive/messaging/fat/kafka/message/ConsumerRecordBean.java

```diff
@@ -0,0 +1,96 @@
+/*******************************************************************************
+ * Copyright (c) 2020 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package com.ibm.ws.microprofile.reactive.messaging.fat.kafka.message;
+
+import java.io.UnsupportedEncodingException;
+
+import javax.enterprise.context.ApplicationScoped;
+
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.common.header.Header;
+import org.eclipse.microprofile.reactive.messaging.Incoming;
+import org.eclipse.microprofile.reactive.messaging.Message;
+import org.eclipse.microprofile.reactive.messaging.Outgoing;
+
+@ApplicationScoped
+public class ConsumerRecordBean {
+
+    public static final String CHANNEL_IN = "consumer-record-in";
+    public static final String CHANNEL_OUT = "consumer-record-out";
+    public static final String GROUP_ID = "consumer-record-app-group";
+
+    public static final String TOPIC = CHANNEL_IN;
+    public static final String KEY = "test_key";
+    public static final String VALUE = "hello";
+    //public static final int PARTITION = 9;
+    public static final long TIMESTAMP = 100L;
+    public static final int NUM_HEADERS = 5;
+    public static final String HEADER_KEY_PREFIX = "headerKey";
+    public static final String HEADER_VALUE_PREFIX = "headerKey";
+    public static final String PASS = "PASS";
+
+    @Incoming(CHANNEL_IN)
+    @Outgoing(CHANNEL_OUT)
+    public Message<String> checkConsumerRecord(Message<String> in) throws UnsupportedEncodingException {
+
+        @SuppressWarnings("unchecked")
+        ConsumerRecord<String, String> consumerRecord = in.unwrap(ConsumerRecord.class);
+
+        String incomingKey = consumerRecord.key();
+        String incomingValue = consumerRecord.value();
+        String incomingTopic = consumerRecord.topic();
+        //int incomingPartition = consumerRecord.partition();
```

Comment:
Should be able to check this now.

Reply:
Not done! But it is harder and I want to do it separately.